src/descriptor/key.rs

```diff
@@ -525,6 +525,53 @@ impl error::Error for ConversionError {
     }
 }
 
+pub trait DescriptorInnerKey {
+    /// The fingerprint of the master key associated with this key, `0x00000000` if none.
+    fn master_fingerprint(&self) -> bip32::Fingerprint;
```

Comment:
In e4d9af484b996d05db261be7406d2af78488e527:

As a separate issue, I think we should return an option here and then upstream we should have a `bip32::Fingerprint::from_opt` function that would do the 0x00000000 thing. Though I'm unsure. Maybe this function is only ever used in contexts where the result needs to be serialized (PSBT, encoding xpubs) in which case users always want the existng behavior.

Reply:
That's sure where I'm using it. 😅